pkg/operator/staticpod/controllers.go

```diff
@@ -138,6 +150,17 @@ func (b *staticPodOperatorControllerBuilder) WithPruning(command []string, stati
 	return b
 }
 
+func (b *staticPodOperatorControllerBuilder) WithGuard(deploymentAssetFunc func(string) []byte, deploymentFile string, pdbAssetFunc func(string) []byte, pdbFile string) Builder {
```

Comment:
The guard logically requires the `infraLister configv1listers.InfrastructureLister`, so add `infraLister configv1listers.InfrastructureLister` as an arg to this method please.